src/server/scripts/Spells/spell_generic.cpp

```diff
@@ -4155,6 +4155,38 @@ class spell_gen_landmine_knockback_achievement : public SpellScriptLoader
     }
 };
 
+class spell_gen_clear_debuffs : public SpellScriptLoader
+{
+public:
+    spell_gen_clear_debuffs() : SpellScriptLoader("spell_gen_clear_debuffs") { }
+
+    class spell_gen_clear_debuffs_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_gen_clear_debuffs_SpellScript);
+
+        void HandleScript(SpellEffIndex /*effIndex*/)
+        {
+            if (Unit* caster = GetHitUnit())
+            {
+                Unit::AuraApplicationMap const& auras = caster->GetAppliedAuras();
+                for (Unit::AuraApplicationMap::const_iterator itr = auras.begin(); itr != auras.end(); ++itr)
+                    if (!itr->second->IsPositive())
+                        caster->RemoveAura(itr->second->GetBase());
```

Comment:
this is not crashsafe

i suggest to use `void Unit::RemoveAppliedAuras(std::function<bool(AuraApplication const*)> const& check)` and check for passive auras too